Shopify/cli
# add predeploy size validation for web pixel extensions

### WHY are these changes introduced?

Fixes: https://github.com/Shopify/ce-customer-behaviour/issues/2261

Our DB for web pixel extensions only stores the script content of size `64KB` and under. We want to enforce this limitation in the CLI now so bigger scripts don't make it to our DB and get truncated.



### WHAT is this pull request doing?



Adding a predeploy validation to check for the size of the script. Right now we're setting it at `64KB`, will turn back to `128KB` once we migrate our DB.

### How to test your changes?

You can clone our web pixel test app: https://github.com/Shopify/web-pixels-test-app

We have a branch that currently sets the output at roughly `90KB`: called [test-over-64-KB](https://github.com/Shopify/web-pixels-test-app/compare/master...test-over-64-KB).

You can clone our app, and checkout that branch! Once you've done that:

1. Create a spin constellation with the partners repo. We have a constellation set up where you can do the following command to have all the necessary repos (partners, extensions, etc):


```
spin up web-pixels-manager:ext-dev
```

2. Once you've done that deploy our app from the CLI! 

```
SHOPIFY_SERVICE_ENV=spin pnpm shopify app deploy --path ../web-pixels-test-app
```

3. You should see an error message like so:


<img width="917" alt="Screenshot 2023-04-13 at 4 23 33 PM" src="https://user-images.githubusercontent.com/19720593/231874584-86b5b266-1b2b-41e7-9e3b-3a0d6bc7ec23.png">

### Post-release steps

NA

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
